### PR TITLE
Add flask socket server

### LINKED ISSUE #226: Add flask socket server

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #226

### CODE REVIEW

File: dialoguekit/platforms/flask_socket_platform.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+@dataclass
+class Response:
+    recipient: str
+    message: Message
+
+
+class FlaskSocketPlatform(Platform):
+    def __init__(self, agent_class: Type[Agent]) -> None:
+        """Represents a platform.
```
Comment (NoB0) [inline dialoguekit/platforms/flask_socket_platform.py:57]: Suggestion to precise that it is a Flask socket.io platform.

File: dialoguekit/platforms/flask_socket_platform.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    def on_feedback(self, data: dict) -> None:
+        """Receives feedback from client.
+
+        Args:
+            data: Data received from client.
+        """
+        req: SocketIORequest = cast(SocketIORequest, request)
+        logger.info(f"Feedback received: {data}")
+        self._platform.feedback(req.sid, **data["feedback"])
+        send({"info": "Feedback received"})
```
Comment (NoB0) [inline dialoguekit/platforms/flask_socket_platform.py:151]: Can you please explain why do we need this line?

**Review by NoB0 — COMMENTED** [review]
Can you add that the Flask socket.io platform is available in the [documentation](https://github.com/iai-group/DialogueKit/blob/main/docs/source/dialogue_orchestration.rst)?

File: dialoguekit/platforms/flask_socket_platform.py (hunk 1 above)
Comment (IKostric) [reply dialoguekit/platforms/flask_socket_platform.py:57]: Done

File: dialoguekit/platforms/flask_socket_platform.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+        Args:
+            namespace: Namespace.
+            platform: An instance of FlaskSocketPlatform.
+        """
+        super().__init__(namespace)
+        self._platform = platform
+
+    def on_connect(self) -> None:
+        """Connects client to server.
```
[the remark this answers is not in the document]
Comment (IKostric) [reply dialoguekit/platforms/flask_socket_platform.py:117]: Done

File: dialoguekit/platforms/flask_socket_platform.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+        Args:
+            data: Data received from client.
```
[the remark this answers is not in the document]
Comment (IKostric) [reply dialoguekit/platforms/flask_socket_platform.py:120]: Done

File: dialoguekit/platforms/flask_socket_platform.py (hunk 2 above)
Comment (IKostric) [reply dialoguekit/platforms/flask_socket_platform.py:151]: Used for verification that the message was recieved. It is not used on the client side. Removed.

**Review by NoB0 — APPROVED** [review]
LGTM

**Pull request merged**